devito/types/dimension.py

```diff
@@ -676,22 +676,24 @@ class ConditionalDimension(DerivedDimension):
     is_NonlinearDerived = True
     is_Conditional = True
 
-    def __new__(cls, name, parent, factor=None, condition=None, indirect=False):
+    def __new__(cls, name, parent, factor=None, condition=None, indirect=False,
+                spacing=None):
```

Comment:
the docstring needs to be updated. You can probably use the ``spacing`` definition from other docstrings in this file

Reply:
though you also have to say optional -- defaults to parent's